flink-table/flink-table-api-java/src/main/java/org/apache/flink/table/catalog/CatalogTableImpl.java

```diff
@@ -32,65 +32,39 @@
 import static org.apache.flink.util.Preconditions.checkNotNull;
 
 /**
- * An abstract catalog table.
+ * A catalog table implementation.
  */
-public class CatalogTableImpl implements CatalogTable {
-	// Schema of the table (column names and types)
-	private final TableSchema tableSchema;
-	// Partition keys if this is a partitioned table. It's an empty set if the table is not partitioned
-	private final List<String> partitionKeys;
-	// Properties of the table
-	private final Map<String, String> properties;
-	// Comment of the table
-	private final String comment;
+public class CatalogTableImpl extends AbstractCatalogTable {
+
+	// The path of the table in its catalog.
+	private final ObjectPath tablePath;
 
 	public CatalogTableImpl(
-		TableSchema tableSchema,
-		Map<String, String> properties,
-		String comment) {
-		this(tableSchema, new ArrayList<>(), properties, comment);
+			ObjectPath tablePath,
+			TableSchema tableSchema,
+			Map<String, String> properties,
+			String comment) {
+		this(tablePath, tableSchema, new ArrayList<>(), properties, comment);
 	}
 
 	public CatalogTableImpl(
+			ObjectPath tablePath,
```

Comment:
If a CatalogTable can have an ObjectPath field, do we need some validation in APIs like `HiveCatalog::createTable`?
And btw, why did we decide to separate CatalogTable and ObjectPath in the first place?

Reply:
I added some validation.

As to separation, CatalogTable as API doesn't require ObjectPath, as it's consistent how objects are organized in databases. However, as an impl class such as CatalogTableImpl, it can has anything it needs without violating the API. One implementation might not need this, and we don't want to enforce this at the API level.